### DISCUSSION TITLE
ESP32-S2

### DISCUSSION THREAD

**Discussion by nikthefix** [Bodmer/TFT_eSPI#859] [Compatible displays and setup files]
[2020-12-12 10.07.35.mov.zip](https://github.com/Bodmer/TFT_eSPI/files/5682861/2020-12-12.10.07.35.mov.zip)

Hi Bodmer,

Here's a short video of TFT_eSPI the ESP32-S2,
Wrover with 30 sprites in psram.
GC9A01 driver @ 240x240 resolution.
Using latest Arduino esp32-s2 core branch.
I was able to achieve a frame rate many times higher than shown here with no problems.

Modifications I had to make:

#ifdef ESP32 to #ifdef ESP32-S2
VSPI references to FSPI
SetSwapBytes tweaks as these settings were different for ESP32 for reasons I have yet to discover (but only for displaying spites as all other drawing functions are the same). 

Interestingly, I found that I had to use 'pushRotated(0)' for all sprite draws as I couldn't as yet get correct colours with 'pushSprite' no matter how I tweaked the SetSwapBytes function.  

Thanks 

Nick

**Comment by Bodmer** [comment]
The new single core ESP32-S2 is not currently supported by the library and the board package for that processor has not been released yet. To avoid nugatory effort I plan to wait until the board package is released, then after the release has stabilised I will start looking at supporting that processor.
The library uses a mix of library calls and low level register access, the low level access functions are perhaps not compatible with the new processor.

**Reply by Startech-tuncer** [reply]
Yeni tek çekirdekli ESP32-S2 de TFT_eSPI çok net çalışıyor.Tabiki 1-2 değişiklik yaptım.

**Comment by zsoltmar** [comment]
I'm curious if there are any updates regarding the ESP32-S2. Cheers!

**Comment by Startech-tuncer** [comment]
Not yet

iPhone’umdan gönderildi

**Comment by Startech-tuncer** [comment]
but it's a great processor , anyone who knows how to use it write if you need help 

iPhone’umdan gönderildi

**Comment by Bodmer** [comment]
I'll have to order a ESP32-S2 board, any recomendations for which one to buy?

**Comment by nikthefix** [comment]
Hi Bodmer,

For development and debugging I recommend the nanoESP32-S2. It has the same layout as the official Espressif Saola dev board but also breaks out the USB port along with the serial chip, both on type C connectors. The wrover variant has added psram:

https://github.com/wuxx/nanoESP32-S2/blob/master/README_en.md

https://www.aliexpress.com/item/1005001403045642.html?src=google&src=google&albch=shopping&acnt=494-037-6276&slnk=&plac=&mtctp=&albbt=Google_7_shopping&albagn=888888&isSmbAutoCall=false&needSmbHouyi=false&albcp=1622677378&albag=57224001810&trgt=743612850714&crea=en1005001403045642&netw=u&device=c&albpg=743612850714&albpd=en1005001403045642&gclid=Cj0KCQjwt-6LBhDlARIsAIPRQcLIHlZz5aN3kI-EzRcdbDCcg0W8dakpN2YraZA-ga1lvmvLclc1i4waAhAuEALw_wcB&gclsrc=aw.ds&aff_fcid=d765fe4740b044dd83356b1b2f35db56-1635574085623-07435-UneMJZVf&aff_fsk=UneMJZVf&aff_platform=aaf&sk=UneMJZVf&aff_trace_key=d765fe4740b044dd83356b1b2f35db56-1635574085623-07435-UneMJZVf&terminal_id=fd99687cce854e43922f67e150d1c649

For a smaller footprint and very low price then the Wemos / Lolin S2mini and S2pico are a good choice but there is no usb-serial chip so you will do your uploads and debugging using usb cdc. This currently works well but does not at present support the deepest debug outputs of the esprssive sdk. Fine for printf etc. Links here:

https://www.wemos.cc/en/latest/s2/s2_mini.html
https://www.wemos.cc/en/latest/s2/s2_pico.html

https://www.aliexpress.com/item/1005003145192016.html?spm=a2g0o.productlist.0.0.48fbb0f5tyifjY&algo_pvid=f0ca8aed-517d-4c79-b2fb-3736b6867865&algo_exp_id=f0ca8aed-517d-4c79-b2fb-3736b6867865-1&pdp_ext_f=%7B%22sku_id%22%3A%2212000024338777694%22%7D

https://www.aliexpress.com/item/1005003215673294.html?spm=a2g0o.productlist.0.0.1a9a7526vOrTax&algo_pvid=f90cfd7a-f3c3-45af-a4be-077efeb4d3a4&algo_exp_id=f90cfd7a-f3c3-45af-a4be-077efeb4d3a4-6&pdp_ext_f=%7B%22sku_id%22%3A%2212000024699468119%22%7D

The nanoESP32-S2 is my favourite and is very robust - I ordered 10 upon release and have had no failures.
The Wemos devices use the new ESP32-S2FN4R2 chip which is interesting as it integrates the flash and psram into the S2 silicon.

**Comment by Startech-tuncer** [comment]
the following is appropriate.

https://m.tr.aliexpress.com/item/1005001403045642.html?trace=wwwdetail2mobilesitedetail&src=google&src=google&albch=shopping&acnt=494-037-6276&slnk=&plac=&mtctp=&albbt=Google_7_shopping&albagn=888888&isSmbAutoCall=false&needSmbHouyi=false&albcp=1622677378&albag=57224001810&trgt=743612850714&crea=en1005001403045642&netw=u&device=c&albpg=743612850714&albpd=en1005001403045642&gclid=Cj0KCQjwt-6LBhDlARIsAIPRQcLIHlZz5aN3kI-EzRcdbDCcg0W8dakpN2YraZA-ga1lvmvLclc1i4waAhAuEALw_wcB&gclsrc=aw.ds&aff_fcid=d765fe4740b044dd83356b1b2f35db56-1635574085623-07435-UneMJZVf&aff_fsk=UneMJZVf&aff_platform=aaf&sk=UneMJZVf&aff_trace_key=d765fe4740b044dd83356b1b2f35db56-1635574085623-07435-UneMJZVf&terminal_id=fd99687cce854e43922f67e150d1c649 iPhone’umdan gönderildi

**Comment by mike-rankin** [comment]
I would like to connect a display to the Wemos S2 mini but I'm not sure if there are default/hardware MISO, MOSI and SCLK pins to use. You always have to specify the CS, DC and RST pins but sometime not the hardware SPI pins. I did some digging and it looks like in the Arduino IDE under Boards: ESP32S2 Dev Module and use hardware SPI pins: MISO 37, SCL 36, MOSI 35, SS 34 but I'm not sure.

**Comment by Bodmer** [comment]
@nikthefix, @Startech-tuncer thanks for the suggestions I have ordered a ESP32-S2 board to experiment with and anitcipate it will be relatively easy to update the library.

**Comment by Startech-tuncer** [comment]
your e-spi driver I adapted to esp32-s2 and it works fast. Thanks 

if you want i can send it

iPhone send

**Reply by mike-rankin** [reply]
Hey @Startech-tuncer I'm starting a new ESP32-S2 project and find the TFT_eSPI library to be the best of them all. I'd love a link to your working library.

**Reply by aaga** [reply]
Likewise!
